src/components/admin/AddTask.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react';
+
+class AddTask extends Component {
+  state = {
+    isUploading: false,
+    uploadSuccessful: false,
+    error: null,
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { onTaskAdd } = this.props;
+    const formData = this.getFormData();
+
+    this.setState({
+      isUploading: true,
+      uploadSuccessful: false,
+      error: null,
+    });
+
+    onTaskAdd(formData).then(({ error }) => {
+      if (error) {
+        this.setState({ isUploading: false, error });
+      } else {
+        this.setState({ isUploading: false, uploadSuccessful: true });
+      }
+    });
+  }
+
+  getFormData = () => {
+    const formData = new window.FormData();
+
+    formData.append('name', this.refs.name.value.trim());
```

Comment:
Setting refs per-input to get values is not optimal here: we tend to use refs as a last resort when we absolutely need to interact with the DOM. Form values are best stored in state, using onChange listeners, and you have a variety of options for creating abstractions that let that be non-boilerplatey